### PR TITLE
feat(IRO): IRO Rollapp token fee 

### PR DESCRIPTION
**Pull request by mtsitrin**
## Description

----

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

----

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: x/iro/keeper/create_plan.go [hunk 1]
```diff
@@ -129,6 +130,11 @@ func (k Keeper) CreatePlan(ctx sdk.Context, allocatedAmount math.Int, start, pre
 		return "", err
 	}
 
+	// 1 reserved token amount
```
Comment (omritoptix) [inline x/iro/keeper/create_plan.go:133]:
```suggestion
	// charge rollapp token creation fee. Same as DYM creation fee, will be used to open the pool. 
```

File: x/iro/keeper/create_plan.go [hunk 2]
```diff
@@ -129,6 +130,11 @@ func (k Keeper) CreatePlan(ctx sdk.Context, allocatedAmount math.Int, start, pre
...
+	// FIXME: make const
+	reserve := math.NewIntWithDecimal(1, int(rollapp.GenesisInfo.NativeDenom.Exponent))
```
Comment (omritoptix) [inline x/iro/keeper/create_plan.go:135]: I think the `reserve` concept is confusing/ambigious as rollapp token charging acts exactly as the dym creation fee - charged and being put into a pool. so I'd stick with creation fee (for both dym and rollapp token). 

Also - wouldn't it be worth just adding a const now? what's the effort here?

File: x/iro/keeper/settle.go [hunk 3]
```diff
@@ -86,8 +86,9 @@ func (k Keeper) Settle(ctx sdk.Context, rollappId, rollappIBCDenom string) error
 // - Creates a balancer pool with the determined tokens and DYM.
 // - Uses leftover tokens as incentives to the pool LP token holders.
 func (k Keeper) bootstrapLiquidityPool(ctx sdk.Context, plan types.Plan) (poolID, gaugeID uint64, err error) {
-	unallocatedTokens := plan.TotalAllocation.Amount.Sub(plan.SoldAmt)        // assumed > 0, as we enforce it in the Buy function
-	raisedDYM := k.BK.GetBalance(ctx, plan.GetAddress(), appparams.BaseDenom) // assumed > 0, as we enforce it by IRO creation fee
+	reserve := math.NewIntWithDecimal(1, int(plan.BondingCurve.SupplyDecimals()))
```
Comment (omritoptix) [inline x/iro/keeper/settle.go:89]: are we guranteed that `plan.BondingCurve.SupplyDecimals` will always be = `rollapp.GenesisInfo.NativeDenom.Exponent` ? 
as the first is used to calcualte the rollapp token fee on the liquidity pool and the latter upon creation. 

the `reserve` here - same comment as before about referring it to rollapp creation fee.

**Review by omritoptix — COMMENTED** [review]
looks good.

mostly cosmetics/naming/questions.

File: x/iro/keeper/settle.go (hunk 3 above)
Comment (mtsitrin) [reply x/iro/keeper/settle.go:89]: `rollapp.GenesisInfo.NativeDenom.Exponent`  supports only 18
there's TODO to add support if needed

Comment (mtsitrin) [reply x/iro/keeper/settle.go:89]: renamed

**Review by omritoptix — APPROVED** [review]

**Pull request merged**